GoogleForCreators/web-stories-wp
# Context Menu: New elements should be selected after duplication

## Context

Issue found during Bug Bash 1.16.0.

## Summary

Duplicating one or more elements will select the newly created element(s).

## Relevant Technical Choices



## User-facing changes

|Multiple|Single|From Layer Panel|
|--|--|--|
|![duplicateelements](https://user-images.githubusercontent.com/22185279/149976723-b0dfa725-9982-4556-984f-8c6ebd186e0f.gif)|![layerduplicate](https://user-images.githubusercontent.com/22185279/149976997-449ca5d9-ede7-4bd0-9248-0b8365cb320b.gif)|![duplicateelement](https://user-images.githubusercontent.com/22185279/149977020-c9328577-7ed0-4869-87ac-9c3760532eda.gif)|

## Testing Instructions




You will need to test that duplicating elements still works as expected:
- duplicates all animations applied to element
- places newly duplicated element in a different spot

New functionality:
- the newly created element will be selected after duplication.
- clicking the duplication button in the layer panel should select the new element on creation.

This PR can be tested by following these steps:

1. Add element(s) to canvas. Add styles + animations to your heart's content
2. Select element(s)
3. Duplicate
4. Verify that new elements are selected and the old elements are not
5. Verify that elements were duplicated correctly


## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #10179
